src/client/components/SearchMessagesWords/SearchMessages.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import './SearchMessages.css';
+
+export default function SearchMessages() {
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const [searchResults, setSearchResults] = React.useState([]);
+
+  const handleInputChange = (event) => {
+    setSearchTerm(event.target.value);
+  };
+
+  useEffect(() => {
```

Comment:
You should merge these two `useEffect` and just fetch new messages from the server whenever a user types something.